bouncy-bullfrogs/bouncy/mail/views.py

```diff
@@ -0,0 +1,189 @@
+from .models import Email, User
+from django.shortcuts import HttpResponse, HttpResponseRedirect, render
+from django.urls import reverse
+import json
+from django.db.models import Q
+from django.contrib.auth import authenticate, login, logout
+from django.contrib.auth.decorators import login_required
+from django.db import IntegrityError
+from django.views.decorators.csrf import csrf_exempt
+from django.http import JsonResponse
+# Create your views here.
+
+
+def home(request):
+    if request.user.is_authenticated:
+        context = {}
+        query = ""
+        if request.GET:
+            query = request.GET['q']
+            context['query'] = get_mail_queryset(request, query)
+            return render(request, "mail/inbox.html", context)
+        else:
+            return render(request, "mail/inbox.html")
+    else:
+        return HttpResponseRedirect(reverse("login"))
+
+
+def info(request):
+    # for now, it will only have one url link
+    return render(request, "aboutus.html")
+
+
+def get_mail_queryset(request, query=None):
+    if query is not None:
+        queryset = []
+        queries = query.split(" ")
+        user_id = User.objects.get(email=request.user.email).id
+        for q in queries:
+            queryset += list(Email.objects.filter(
```

Comment:
This looks like a very loaded line, and could probably be split into several expressions.